test/test_helper.rb

```diff
@@ -64,3 +64,12 @@ def setup
       instrumentation_service: @service
   end
 end
+
+# Provide information about socket creation
+class SocketHelpers
+  class << self
+    def default_socket_opts
+      { connect_timeout: Net::LDAP::Connection::DefaultConnectTimeout }
+    end
+  end
+end
```

Comment:
Also, it'd be good to not depend on the implementation in the test. So avoid referencing Net::LDAP::Connection::DefaultConnectTimeout in the test, and use the actual value instead. If that changes, then it's easy to see where the failures are and replace them.